src/helpers/simplify_claims.ts

```diff
@@ -1,176 +1,166 @@
+    valueObj = { value } as CustomSimplifiedClaim
+  }
 
   if (keepRanks) valueObj.rank = rank
 
-  const subSnaksOptions = getSubSnakOptions(options)
-  subSnaksOptions.keepHashes = keepHashes
-
   if (keepQualifiers) {
-    valueObj.qualifiers = simplifyQualifiers(claim.qualifiers, subSnaksOptions)
+    valueObj.qualifiers = simplifyQualifiers(claim.qualifiers, options)
   }
 
   if (keepReferences) {
     claim.references = claim.references || []
-    valueObj.references = simplifyReferences(claim.references, subSnaksOptions)
+    valueObj.references = simplifyReferences(claim.references, options)
   }
 
   if (keepIds) valueObj.id = claim.id
 
   return valueObj
 }
 
-export function simplifyClaims (claims: Claims, options: SimplifySnaksOptions = {}): SimplifiedClaims {
-  return simplifySnaks(claims, options)
+export function simplifyClaims (claims: Claims, options: SimplifyClaimsOptions = {}): SimplifiedClaims {
+  return applyObjectSimplification(claims, simplifyPropertyClaims, options)
 }
-export function simplifyPropertyClaims (propertyClaims: PropertyClaims, options: SimplifySnaksOptions = {}): SimplifiedPropertyClaims {
-  return simplifyPropertySnaks(propertyClaims, options)
+
+export function simplifyPropertyClaims (propertyClaims: PropertyClaims, options: SimplifyClaimsOptions = {}): SimplifiedPropertyClaims {
+  // Avoid to throw on empty inputs to allow to simplify claims array
+  // without having to know if the entity as claims for this property
+  // Ex: simplifyPropertyClaims(entity.claims.P124211616)
+  if (propertyClaims == null || propertyClaims.length === 0) return []
+
+  const { keepNonTruthy, keepNonDeprecated } = options
+
+  if (keepNonDeprecated) {
+    propertyClaims = nonDeprecatedPropertyClaims(propertyClaims)
+  } else if (!(keepNonTruthy)) {
+    propertyClaims = truthyPropertyClaims(propertyClaims)
+  }
+
+  return applyArraySimplification(propertyClaims, simplifyClaim, options)
 }
-export function simplifyClaim (claim: Claim, options: SimplifySnakOptions = {}): SimplifiedClaim {
-  return simplifySnak(claim, options)
+
+export function simplifySnaks (snaks: Snaks = {}, options: SimplifySnaksOptions = {}): SimplifiedSnaks {
+  return applyObjectSimplification(snaks, simplifyPropertySnaks, options)
+}
+
+export function simplifyPropertySnaks (propertySnaks: PropertySnaks, options: SimplifySnaksOptions = {}): SimplifiedPropertySnaks {
+  if (propertySnaks == null || propertySnaks.length === 0) return []
+  return applyArraySimplification(propertySnaks, simplifySnak, options)
+}
+
+function applyObjectSimplification (obj, fn, options) {
+  const { propertyPrefix } = options
+  const simplified = {}
+  for (let [ propertyId, propertyArray ] of Object.entries(obj)) {
+    if (propertyPrefix) {
+      propertyId = propertyPrefix + ':' + propertyId
+    }
+    simplified[propertyId] = fn(propertyArray, options)
+  }
+  return simplified
+}
+
+function applyArraySimplification (array, simplifyFn, options) {
```

Comment:
provide typings to notice mistakes
```suggestion
function applyArraySimplification<
  Input,
  Output,
  Options,
  F extends (element: Input, options: Options) => Output,
>(
  array: readonly Input[],
  simplifyFn: F,
  options: Options,
): Output[] {
```